cmd/jujud/agent/agenttest/engine.go

```diff
@@ -187,3 +188,29 @@ func (tracker *EngineTracker) startFunc(id string, names []string) dependency.St
 		return nil, dependency.ErrMissing
 	}
 }
+
+// AssertManifoldsDependencies asserts that given manifolds have expected dependencies.
+func AssertManifoldsDependencies(c *gc.C, manifolds dependency.Manifolds, expected map[string][]string) {
+	dependencies := make(map[string][]string, len(manifolds))
+	manifoldNames := set.NewStrings()
+
+	for name, manifold := range manifolds {
+		manifoldNames.Add(name)
+		dependencies[name] = ManifoldDependencies(manifolds, name, manifold).SortedValues()
+	}
+	c.Assert(len(dependencies), gc.Equals, len(expected))
+
+	for _, n := range manifoldNames.SortedValues() {
+		c.Assert(dependencies[n], jc.SameContents, expected[n])
+	}
+}
+
+// ManifoldDependencies returns all manifold dependencies.
```

Comment:
Maybe clarify that "all" means all direct and indirect dependencies here? It's a bit ambiguous - could just mean all the direct dependencies including the implicit ones from various decorations.